zqd/space/space.go

```diff
@@ -34,59 +35,199 @@ var (
 	ErrSpaceNotExist       = zqe.E(zqe.NotFound, "space does not exist")
 )
 
+type Manager struct {
+	rootPath string
+	spaces   sync.Map
+}
+
 type Space struct {
 	path string
 	conf config
+
+	// state about operations in progress
+	opMutex       sync.Mutex
+	active        int
+	deletePending bool
+
+	wg sync.WaitGroup
+	// closed to signal non-delete ops should terminate
+	cancelChan chan struct{}
 }
 
-func Open(root, name string) (*Space, error) {
-	path := filepath.Join(root, name)
-	c, err := loadConfig(path)
+func NewManager(root string) *Manager {
```

Comment:
I suggest NewManager takes a zap.Logger instance that is stored on the Manager (and used at the "log somewhere" comments), and that NewManager returns `(*Manager, error)`.